logzio-sender/src/main/java/io/logz/sender/HttpsSyncSender.java

```diff
@@ -121,33 +92,67 @@ public void sendToLogzio(List<FormattedLogMessage> messages) throws LogzioServer
                     reporter.error("Got IO exception - " + e.getMessage());
                 }
 
-                if (!shouldRetry && responseCode == HttpURLConnection.HTTP_OK) {
-                    reporter.info("Successfully sent bulk to logz.io, size: " + payload.length);
-                    break;
-
-                } else {
-
-                    if (currTry == configuration.getMaxRetriesAttempts()){
-
+                if (shouldRetry) {
+                    if (currTry == configuration.getMaxRetriesAttempts()) {
```

Comment:
I must be tired, but the previous code is complicated, I can't track the logic. Without a notebook and a pen I can't write down the algorithm. Maybe refactor it a bit?

Reply:
@asafm Yes, it was very complex and I tried not to touch it because this is a sensitive project. I'll refactor a bit more tomorrow